app/forms/batch_upsert_holidays.rb

```diff
@@ -0,0 +1,63 @@
+class BatchUpsertHolidays
+  include ActiveModel::Model
+  extend ActiveModel::Naming
+  include DateRangePickerHelper
+
+  attr_reader :title
+  attr_reader :date_range
+  attr_reader :users
+  attr_reader :all_day
+
+  validates :title, presence: true
+  validates :date_range, presence: true
+  validates :users, presence: true
+
+  def to_model
+    if @previous_holiday_ids.any?
+      @previous_holiday ||= Holiday.find(@previous_holiday_ids.first)
+    else
+      Holiday.new
+    end
+  end
+
+  def initialize(previous_holiday_ids = [], title = nil, all_day = true, date_range = nil, users = [])
+    @previous_holiday_ids = previous_holiday_ids
+    @title = title
+    @all_day = ActiveRecord::Type::Boolean.new.cast(all_day)
+    @date_range = date_range
+    @users = users
+  end
+
+  def call
+    return false unless valid?
+    Holiday.where(id: @previous_holiday_ids).destroy_all
+    start_at, end_at = calculate_date_range
+    create_holidays(start_at, end_at)
```

Comment:
FWIW I feel the intermediary assignments of the start / end dates are not necessary. I'd favour nesting the call such as `create_holidays(calculate_date_range)`.